packages/stockflux-launcher/src/App.js

```diff
@@ -1,66 +1,144 @@
-import React, { useRef, useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import cx from 'classnames';
-import { OpenfinApiHelpers } from 'stockflux-core';
-import { useDockWindow, ScreenEdge } from 'openfin-react-hooks';
+import { useDockWindow, useOptions, ScreenEdge } from 'openfin-react-hooks';
 import {
   FaChevronUp,
   FaChevronLeft,
   FaChevronRight,
-  FaRegHandRock
+  FaRegHandRock,
+  FaUnlock
 } from 'react-icons/fa';
-import AppShortcuts from './app-shortcuts/AppShortcuts';
 import FreeTextSearch from './free-text-search/FreeTextSearch';
 import ToolBar from './toolbar/ToolBar';
 import CloseButton from './toolbar/CloseButton';
 import './App.css';
 
-const isLauncherHorizontal = edge => edge === ScreenEdge.TOP;
-
 export default () => {
+  const [alignment, setAlignment] = useState('horizontal');
+  const [isCloudMode, setCloudMode] = useState(false);
+  const [isDocked, setDocked] = useState(true);
+  const [options] = useOptions();
+  const [undockWidth, setUndockWidth] = useState(1000);
+  const [undockHeight, setUndockHeight] = useState(50);
+  const [undockTop] = useState(50);
+  const [undockLeft, setUndockLeft] = useState(0);
+
   const [edge, windowActions] = useDockWindow(
     ScreenEdge.TOP,
-    OpenfinApiHelpers.getCurrentWindowSync(),
+    window.fin.Window.getCurrentSync(),
     true,
-    { dockedWidth: 50, dockedHeight: 50 }
+    { dockedWidth: 50, dockedHeight: 50 },
+    {
+      undockPosition: { top: undockTop, left: undockLeft },
+      undockSize: { width: undockWidth, height: undockHeight }
+    }
   );
 
-  const prevEdgeRef = useRef();
+  let currentDirection = edge !== ScreenEdge.NONE ? edge : ScreenEdge.TOP;
+  let left =
+    window.screenLeft >= window.screen.availWidth ? window.screenLeft : 0;
+  let right =
+    window.screenLeft >= window.screen.availWidth
+      ? window.screenLeft
+      : window.screen.availWidth - 50;
+  let midpoint =
+    left === 0
+      ? (left + right) / 4
+      : window.screen.availWidth + (left + right) / 8;
+
+  // Handle initialDocked false resize to mini window
+  useEffect(() => {
+    if (
+      options &&
+      options.customData &&
+      options.customData.initialDocked === false
+    ) {
+      setUndockLeft(midpoint);
+      windowActions.dockNone();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options]);
```

Comment:
Are we sure this eslint-disable-next-line is needed?

Reply:
I added it here to avoid the effect also watching the window, since I only want the function to fire once - when the options has loaded and been checked (i.e. to undock on startup). I will check if it can be removed.

Reply:
This will not fire once - it should fire every time options change (which should not happen ofc). If you want this to happen only once loaded then an empty array should suffice. I'm happy to approve this, just something to consider.